src/v1/controller/reserve_ticket.ts

```diff
@@ -0,0 +1,337 @@
+
+    // insert ticket in tickets table
+    const amount = calculate_amount(
+      train_id,
+      validate.customers.length,
+      destination_minutes - source_minute
+    );
+    const seatnumber = 1;
+    const userid = 1;
+
+    const { mobile, familymobile, email, district, country, pincode, state } =
+      validate;
+    const ticket_entry_data = {
+      quantity: validate.customers.length,
+      startdate: validate.startdate,
+      enddate,
+      amount,
+      mobile,
+      familymobile,
+      email,
+      status: 1,
+      district,
+      country,
+      pincode,
+      state,
+      pnr,
+      userid: (req as CustomRequest).user.id,
+      source_date: formated_train_date,
+      train_id,
+      start_schedule: source_schedule.id,
+      end_schedule: destination_schedule.id,
+    };
+    console.log("Ticket-entry-data", ticket_entry_data);
+    const response = await reservation.insert_reserve_ticket(ticket_entry_data);
+    console.log("RRRRRrr", response);
+    // insert customers in customer table
+
+    const insert_customer_response = await Promise.all(
+      validate.customers.map(({ first_name, last_name, dob }: any) => {
+        return CUSTOMERS.insert_customers({
+          first_name,
+          last_name,
+          dob: formatDateString(dob, "YYYY-MM-DD HH:mm:ss"),
+        });
+      })
+    );
+    console.log(insert_customer_response);
+
+    // const customer_entry_data = { firstname, lastname, dob };
+    // insert ticket id and customer id in customerticket table
+    const start = count + 1;
+    let all_seat: number[] = [];
+    const insert_customer_ticket = await Promise.all(
+      insert_customer_response.map((id: any, i: number) => {
+        const seat_number = start + i;
+        all_seat.push(seat_number);
+        return CUSTOEMER_TICKET.insert_customer_ticket({
+          customerid: id,
+          ticketid: response,
+          seat_number,
+        });
+      })
+    );
+    console.log(insert_customer_ticket);
+
+    const data = {
+      souce_station_name: source_station.stn_name,
+      source_station_departure_time: arrival_time_at_source_station,
+      start_date_of_journey: validate.startdate,
+      destination_station_name: destination_station.stn_name,
+      seats: all_seat,
+      passengers: validate.customers,
+      end_date_of_journey,
+      arrival_time_at_destination_station,
+      train_number: validate.trainnumber,
+      pnr,
+    };
+
+    res.status(200).json({
+      error: false,
+      data,
+      message: "this is your ticket",
+    });
+    // const customerticket_entry_data = { ticketid, customerid };
+  } catch (error) {
+    const err = error as any;
+    res.status(400).json({
+      error: true,
+      message: err.message,
+      data: {},
+    });
+  }
+};
+
+export const get_all_ticket = async (req: Request, res: Response) => {
+  try {
+    const general_ticket_query = `select * from general_ticker where agentid = '${
```

Comment:
I hope you have tested this API because I couldn't find the table with the name general_ticker, how the response can come from the API.